### PR TITLE
chore: upgrade intellij plugin to remediate compatibility issues

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description
This PR updates the plugin so that incompatibilities with the latest IntelliJ versions are removed. All dependencies are updated, beginning with this plugin version the minimum version is IntelliJ Platform 2023.1. For older platforms, the plugin version 2.4.63 can still be installed.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

image

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/Utils.kt [hunk 1]
```diff
@@ -152,8 +153,8 @@ fun isUrlValid(url: String?): Boolean {
     url.isNullOrEmpty() && return true
```
Comment (asaf92) [inline src/main/kotlin/io/snyk/plugin/Utils.kt:153]: Why is an empty string/null considered a "valid URL"?

File: src/test/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt
Comment (asaf92) [inline src/test/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt:1]: Will these tests still run?

File: src/test/java/ai/deepcode/javaclient/core/DeepCodeIgnoreInfoHolderTest.java [hunk 2]
```diff
@@ -15,12 +15,10 @@
 
 public class DeepCodeIgnoreInfoHolderTest {
 
-  private final File basicIgnoreFile =
-    new File(getClass().getClassLoader().getResource("basic/.dcignore").getPath());
+  private final File basicIgnoreFile = new File("src/test/resources/basic/.dcignore");
```
Comment (asaf92) [inline src/test/java/ai/deepcode/javaclient/core/DeepCodeIgnoreInfoHolderTest.java:18]: The previous way looks safer imo

**Review by asaf92 — COMMENTED** [review]
What are the implications of all the removed `@Test` annotations?

**Review by asaf92 — APPROVED** [review]

File: src/main/kotlin/snyk/container/ContainerService.kt [hunk 3]
```diff
@@ -74,7 +74,7 @@ class ContainerService(project: Project) : CliAdapter<ContainerIssuesForImage, C
             // jenkins/jenkins      -> jenkins/jenkins/jenkins
             // jenkins/jenkins:lts  -> jenkins/jenkins:lts/jenkins
             // so for `jenkins/jenkins:lts/jenkins` we should choose latter (longest match)
-            nameCandidates.size > 1 -> nameCandidates.maxBy { it.length }!!
+            nameCandidates.size > 1 -> nameCandidates.maxByOrNull { it.length }!!
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/snyk/container/ContainerService.kt:77]: Changes in the Kotlin stdlib

File: src/main/kotlin/io/snyk/plugin/Utils.kt [hunk 4]
```diff
@@ -152,8 +153,8 @@ fun isUrlValid(url: String?): Boolean {
...
 
     return try {
-        val uri = URL(url).toURI()
-        return uri.isSnykTenant()
+        val uri = url?.let { URI.create(it) }
+        return uri?.isSnykTenant() ?: false
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/Utils.kt:157]: The compiler / linter didn't get that, though.

File: src/test/java/ai/deepcode/javaclient/core/DeepCodeIgnoreInfoHolderTest.java (hunk 2 above)
Comment (bastiandoetsch) [reply src/test/java/ai/deepcode/javaclient/core/DeepCodeIgnoreInfoHolderTest.java:18]: But didn't work anymore (on my computer).

File: src/test/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt
Comment (bastiandoetsch) [reply src/test/kotlin/io/snyk/plugin/cli/ConsoleCommandRunnerTest.kt:1]: Yes, as they are inheriting from LightPlatformTestCase, which is JUnit3 (no annotations).

File: src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt [hunk 5]
```diff
@@ -124,7 +125,7 @@ fun getReadOnlyClickableHtmlJEditorPaneFixedSize(
     // don't remove that!
     // Some magic (side-effect? customStyleSheet?) happens when JBHtmlEditorKit() initializing
     // that make html tags like <em>, <p>, <ul> etc. be treated properly inside JEditorPane
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt:127]: still valid - the HTMLEditorKit initialization still initializes the CSS apparently.

**Pull request merged**